### PR TITLE
docs: added information about which pods will have multus annotations

### LINKED ISSUE #7563: Operator pod is missing multus nework annotation with external -> enabled: true
* Bug Report

**Deviation from expected behavior:**
The operator is using the pod network to contact the external ceph cluster instead of the specified multus network.

**Expected behavior:**
Operator deployment template including the multus network annotation.

**How to reproduce it (minimal and precise):**
Try to use rook to access an external cluster trough a multus network. Example configuration:
```
cat <<EOY | kubectl --context=isc apply -f -
apiVersion: ceph.rook.io/v1
kind: CephCluster
metadata:
  name: rook-ceph
  namespace: rook-ceph
spec:
  external:
    enable: true
  network:
    provider: multus
    selectors:
      public: default/my-storage-network
  crashCollector:
    disable: true
  dataDirHostPath: /var/lib/rook
  cephVersion:
    image: ceph/ceph:v15.2.9
EOY
```

This has an indirect connection with https://github.com/rook/rook/issues/7550 together (ceph-mgr cannot be exposed for external access).

**Environment**:
Rook version (use rook version inside of a Rook Pod): 1.5.9
Storage backend version (e.g. for ceph do ceph -v): 15.2.9
Kubernetes version (use kubectl version): v1.20.2
Storage backend status (e.g. for Ceph use ceph health): HEALTH_OK

### PR DESCRIPTION
**Pull request by rohan47**
**Description of your changes:**
Added information about which pods will have multus annotations.

**Which issue is resolved by this Pull Request:**
Resolves #7563 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: Documentation/ceph-cluster-crd.md [hunk 1]
```diff
@@ -340,6 +340,9 @@ Based on the configuration, the operator will do the following:
 
 \* Internal cluster traffic includes OSD heartbeats, data replication, and data recovery
 
+Only OSD pods will have Public as well as Cluster networks attached. Rest of the Ceph component pods and CSI pods will have Only Public network attached.
+Rook Ceph Operator will not have any networks attached.
```
Comment (travisn) [inline Documentation/ceph-cluster-crd.md:344]: Does this mean the operator is not expected to need any networks attached? Then how does the operator connect to the cluster to run ceph commands? I'm just not clear how all the networks are connected.

File: Documentation/ceph-cluster-crd.md [hunk 2]
```diff
@@ -340,6 +340,9 @@ Based on the configuration, the operator will do the following:
 
 \* Internal cluster traffic includes OSD heartbeats, data replication, and data recovery
 
+Only OSD pods will have Public as well as Cluster networks attached. Rest of the Ceph component pods and CSI pods will have Only Public network attached.
```
Comment (travisn) [inline Documentation/ceph-cluster-crd.md:343]:
```suggestion
Only OSD pods will have both Public and Cluster networks attached. The rest of the Ceph component pods and CSI pods will only have the Public network attached.
```

**Review by travisn — CHANGES_REQUESTED** [review]

**Comment by leseb** [comment]
@rohan47 what's the update here?

**Comment by rohan47** [comment]
> @rohan47 what's the update here?

Updated the PR

File: Documentation/ceph-cluster-crd.md [hunk 3]
```diff
@@ -340,6 +340,9 @@ Based on the configuration, the operator will do the following:
 
 \* Internal cluster traffic includes OSD heartbeats, data replication, and data recovery
 
+Only OSD pods will have both Public and Cluster networks attached. The rest of the Ceph component pods and CSI pods will only have the Public network attached.
+Rook Ceph Operator will not have any networks attached as it proxies the required commands via a [sidecar container](https://github.com/rook/rook/pull/8272) in the mgr pod.
```
Comment (travisn) [inline Documentation/ceph-cluster-crd.md:344]: I'd suggest no need to link the PR
```suggestion
Rook Ceph Operator will not have any networks attached as it proxies the required commands via a sidecar container in the mgr pod.
```

File: Documentation/ceph-cluster-crd.md [hunk 4]
```diff
@@ -379,6 +382,10 @@ spec:
   * This format is required in order to use the NetworkAttachmentDefinition across namespaces.
   * In Openshift, to use a NetworkAttachmentDefinition (NAD) across namespaces, the NAD must be deployed in the `default` namespace. The NAD is then referenced with the namespace: `default/rook-public-nw`
 
+#### Known issues with multus:
```
Comment (travisn) [inline Documentation/ceph-cluster-crd.md:385]:
```suggestion
#### Known issues with multus
```

**Review by travisn — CHANGES_REQUESTED** [review]

**Review by travisn — APPROVED** [review]

**Pull request merged**